### PR TITLE
Added Pester, ShouldProcess and Calculated Property PS Snippets

### PR DESCRIPTION
**Pull request by brettmillerb**
## PR Summary
Added snippets for the following:
* Calculated Property for use with `Select-Object -Property`
* `If Should Process` for quickly adding `if ($PSCmdlet.ShouldProcess("Target", "Operation")) {...}`
* Pester:
    * Describe Block
    * Describe with nested context & It block
    * Context Block
    * Context with nested It Block
    * It Block

## PR Checklist

Note: Tick the boxes below that apply to this pull request by putting an `x` between the square brackets.
Please mark anything not applicable to this PR `NA`.

- [x] PR has a meaningful title
- [x] Summarized changes
- [ ] PR has tests
- [x] This PR is ready to merge and is not work in progress
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready

### CODE REVIEW

File: snippets/PowerShell.json [hunk 1]
```diff
@@ -952,5 +952,72 @@
...
+            "}"
+        ],
+        "description": "Adds ShouldProcess block"
+    },
+    "CalculatedProperty": {
+        "prefix": "Calculated-Property",
+        "body": [
+            "@{name='${1:PropertyName}';expression={\\$_.${2:PropertyValue}}}$0"
+        ],
+        "description": "Creates a PSCustomObject"
```
Comment (TylerLeonhardt) [inline snippets/PowerShell.json]: I thought there was already a `PSCustomObject` snippet in this file. Is there not?

Also, isn't this just a `hashtable`, not a true `PSCustomObject`

Comment (brettmillerb) [reply snippets/PowerShell.json]: Woops. Thats my lazy copy paste it's a calculated property for use with select-object.

I'll update when back on the laptop.

Comment (brettmillerb) [reply snippets/PowerShell.json]: Updated the description for the calculated property @TylerLeonhardt

Comment (TylerLeonhardt) [reply snippets/PowerShell.json]: Sorry can you give me an example of this? I'm having a hard time seeing the scenario.

Comment (brettmillerb) [reply snippets/PowerShell.json]: Linked from the `Select-Object` docs:

https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/select-object?view=powershell-6#examples

```markdown
-Property

Specifies the properties to select. These properties are added as NoteProperty members to the output objects. Wildcards are permitted.

The value of the Property parameter can be a new calculated property. To create a calculated, property, use a hash table. Valid keys are:

    Name (or Label) <string>
    Expression <string> or <script block>
```

```powershell
Get-Process | Select-Object -Property ProcessName,@{Name="Start Day"; Expression = {$_.StartTime.DayOfWeek}}

ProcessName  StartDay
----         --------
alg          Wednesday
ati2evxx     Wednesday
ati2evxx     Thursday
...
```

Comment (brettmillerb) [reply snippets/PowerShell.json]: I'd say it's used quite a bit when working with Active Directory.

```powershell
Get-ADUser -Identity brettmillerb -properties manager | Select-Object -Property @(
    @{name = 'Manager'; expression = { Get-ADUser -identity $_.Manager | Select-Object -ExpandProperty samaccountname }}
)
```

Comment (TylerLeonhardt) [reply snippets/PowerShell.json]: Ahhh I see now! Thanks for the explanation.

File: snippets/PowerShell.json [hunk 2]
```diff
@@ -952,5 +952,72 @@
...
+            "}"
+        ],
+        "description": "Creates ShouldProcess block"
+    },
+    "CalculatedProperty": {
+        "prefix": "Calculated-Property",
+        "body": [
+            "@{name='${1:PropertyName}';expression={\\$_.${2:PropertyValue}}}$0"
+        ],
+        "description": "Creates a Calculated Property"
```
Comment (TylerLeonhardt) [inline snippets/PowerShell.json]:
```suggestion
        "description": "Creates a Calculated Property typically used with Select-Object."
```

**Comment by brettmillerb** [comment]
I updated the calculated property snippet to include the `$_.` and not just the PropertyValue of `$PsItem`

**Comment by TylerLeonhardt** [comment]
Thanks for contributing @brettmillerb!

**Pull request merged**